### PR TITLE
feat: Adds non-interactive code blocks to doc site

### PR DESCRIPTION
**Pull request by zero0Halo**
chore: added prism-react-renderer library, 0.1.7

feat: CodePreview can render code blocks or interactive code

chore: updated existing mdx to use the 'live' prop

fix: removed reference to deleted CodeBlock component

chore: removed test code from modal docs

docs: added comments on the CodePreview component

chore: prettier

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config

---------
**Outline your feature or bug-fix below**

With this feature, we will be able to have non-interactive code blocks in addition to our interactive code blocks. By passing the prop, `live` to an mdx's invocation of a code block, it turns the code block to an interactive code block. By default code blocks are not interactive.

### CODE REVIEW

File: docs/src/components/CodePreview/CodePreview.component.tsx [hunk 1]
```diff
@@ -11,6 +11,9 @@ import {
     LiveError,
 } from 'react-live';
 import Component from '@reach/component-component';
+import Highlight, {defaultProps} from 'prism-react-renderer';
```
Comment (sharkparty) [inline docs/src/components/CodePreview/CodePreview.component.tsx]: I don't think this has been linted.

Comment (sharkparty) [reply docs/src/components/CodePreview/CodePreview.component.tsx]: Even with the ignore, there should be spaces on each side of the `defaultProps` import.

Comment (zero0Halo) [reply docs/src/components/CodePreview/CodePreview.component.tsx]: Odd, I did run `ci`. Not sure why this didn't get picked up. Regardless I fixed it.

**Review by sharkparty — APPROVED** [review]
LGTM

**Pull request merged**